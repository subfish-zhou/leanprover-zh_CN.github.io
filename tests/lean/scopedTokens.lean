namespace Foo

scoped syntax "foo!" term:max : term

macro_rules
  | `(foo! $x) => `($x + 1)

#check foo! 20

theorem ex1 : foo! 20 = 21 := rfl

end Foo

#check foo! 10 -- Error

def foo! := 10

theorem ex2 : foo! = 10 := rfl

#check foo!

open Foo

#check foo! 20

theorem ex3 : foo! 10 = 11 := rfl

namespace Bla

scoped syntax "bla!" term:max : term

macro_rules
  | `(bla! $x) => `($x * 2)

theorem ex2 : bla! 3 = 6 := rfl

end Bla

def bla! := 20

theorem ex4 : bla! = 20 := rfl
