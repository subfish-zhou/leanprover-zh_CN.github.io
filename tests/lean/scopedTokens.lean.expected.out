20 + 1 : Nat
scopedTokens.lean:14:7-14:11: error: unknown identifier 'foo!'
foo! : Nat
20 + 1 : Nat
